core/src/main/java/org/apache/accumulo/core/clientImpl/Tables.java

```diff
@@ -95,28 +92,14 @@ public static TableId getTableId(ClientContext context, String tableName)
     }
   }
 
-  /**
-   * Return the cached ZooCache for provided context. ZooCache is initially created with a watcher
-   * that will clear the TableMap cache for that instance when WatchedEvent occurs.
-   */
   private static ZooCache getZooCache(final ClientContext context) {
     SecurityManager sm = System.getSecurityManager();
     if (sm != null) {
       sm.checkPermission(TABLES_PERMISSION);
     }
 
-    final String uuid = context.getInstanceID();
-
-    try {
-      return instanceToZooCache.get(uuid, () -> {
-        final String zks = context.getZooKeepers();
-        final int timeOut = context.getZooKeepersSessionTimeOut();
-        return new ZooCacheFactory().getZooCache(zks, timeOut,
-            watchedEvent -> instanceToMapCache.invalidate(uuid));
```

Comment:
I think this is very important, it clears the map when zoocache changes.

Reply:
Ugh right.  The TableMap in another process won't get updated.

Reply:
I opened #973 as an alternate way to do this.   It was too hard to explain and easier to write the code.